add docs and local k8s deployment files (#17)
kbase/cdm-kube-spark-manager: README.md

> **⚠️ Important:** This API service is intended to be deployed only within a Kubernetes environment.

## Features

- Create and manage Spark clusters for individual users
- RESTful API endpoints for cluster management
- Automatic cleanup of resources
- KBase user authentication and authorization

## Setup Local Development Environment

### Prerequisites

- Python 3.11+
- pip

### Installation

1. Clone the repository:
   ```
   git clone git@github.com:kbase/cdm-kube-spark-manager.git
   cd cdm-kube-spark-manager
   ```

2. Install dependencies:
   ```
   pip install -r requirements.txt
   ```

## Local Unit Testing

The project uses pytest for unit testing. To run the tests:

```bash
# Run all tests
PYTHONPATH=src pytest -v --cov=src --cov-report=xml tests
```

## API Endpoints

### Health Check
- `GET /health` - Check the health status of the service

### Cluster Management
- `POST /clusters` - Creates a new Spark cluster for the authenticated user

> - `GET /clusters` - List the status of all Spark clusters for the authenticated user

Reviewer: ```suggestion
- `GET /clusters` - Get the status of the Spark cluster for the authenticated user, if it exists
```
